docs/api.md

```diff
@@ -200,6 +200,13 @@ Indicates that the browser is connected.
     - `deviceScaleFactor` <[number]> Specify device scale factor (can be thought of as dpr). Defaults to `1`.
     - `isMobile` <[boolean]> Whether the `meta viewport` tag is taken into account and touch events are enabled. Defaults to `false`. Not supported in Firefox.
   - `userAgent` <?[string]> Specific user agent to use in this context.
+  - `device` <[string]|[Object]> Pass a device name or object from [playwright.devices](#playwrightdevices) to set the viewport and user agent.
+    - `viewport` <?[Object]> Sets a consistent viewport for each page. Defaults to an 1280x720 viewport. `null` disables the default viewport.
```

Comment:
Somewhat related: I think that `isMobile` or `deviceScaleFactor` have nothing to do with viewport. I'd go with separate `viewportSize` and `userAgent` options, and a device option which has everything. WDYT?

Reply:
Yeah, I can also see that as an improvement.

Reply:
Maybe rely upon destructuring here? I.e. leave `viewportSize`, `isMobile`, `deviceScaleFactor` on the top level and allow creating context as 

`browser.newContext({ device: 'iPhone' })`

and

`browser.newContext({...iPhone, })`

?

Reply:
That doesn’t work well with typescript, because device contains ‘name’ and so I believe typescript will yell at you.